tests/orm/nodes/data/test_kpoints.py

```diff
@@ -81,3 +84,79 @@ def test_get_kpoints(self):
         kpt2 = load_node(kpt.pk)
         assert np.abs(kpt2.get_kpoints() - np.array(kpoints)).sum() == 0.0
         assert np.abs(kpt2.get_kpoints(cartesian=True) - np.array(cartesian_kpoints)).sum() == 0.0
+
+
+@skip_atomistic
+class TestKpointsAtomisticStructureData:
+    """Test for the `Kpointsdata` class using the new atomistic StructureData."""
+
+    @pytest.fixture(autouse=True)
+    def init_profile(self):
```

Comment:
I think here you want to create a structure to be shared for many sub-tests. It would be better to use fixture.
The profile is more suitable for if you need to test the store and read data nodes from DB. If you need profile, there is `aiida_profile` fixture to use.

Reply:
Hi @unkcpz, would it be better to put also the last two tests in the above `TestKpoints` class but marked with the skip_atomistic fixture, and then in the init profile add a self.atomistic = self.structure.to_atomistic which run only if has_atomistic?